### ISSUE TITLE
SLATE does not work with Argyris

### ISSUE THREAD

**Issue by pefarrell** [firedrakeproject/firedrake#1310]
Consider

```
from firedrake import *

mesh = UnitSquareMesh(2, 2)
V = FunctionSpace(mesh, "Argyris", 5)

u = TrialFunction(V)
v = TestFunction(V)

a = Tensor(inner(u, v)*dx).inv
M = assemble(a)
M.force_evaluation()
```

This crashes with

```
Traceback (most recent call last):
  File "slateargyris.py", line 11, in <module>
    M.force_evaluation()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/firedrake/firedrake/matrix.py", line 242, in force_evaluation
    self._M._force_evaluation()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/base.py", line 1463, in _force_evaluation
    _trace.evaluate(reads, writes)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/base.py", line 193, in evaluate
    comp._run()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/base.py", line 3986, in _run
    return self.compute()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/base.py", line 4027, in compute
    fun = self._jitmodule
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/utils.py", line 62, in __get__
    obj.__dict__[self.__name__] = result = self.fget(obj)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/sequential.py", line 927, in _jitmodule
    pass_layer_arg=self._pass_layer_arg)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/caching.py", line 199, in __new__
    obj = make_obj()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/caching.py", line 189, in make_obj
    obj.__init__(*args, **kwargs)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/sequential.py", line 756, in __init__
    self.compile()
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/sequential.py", line 842, in compile
    comm=self.comm)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/compilation.py", line 442, in load
    dll = compiler.get_so(src, extension)
  File "/home/pefarrell/local/firedrake/firedrake-dev-20180914/src/PyOP2/pyop2/compilation.py", line 284, in get_so
    Compile errors in %s""" % (e.cmd, e.returncode, logfile, errfile))
```

with compiler error output

```
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:216:0: warning: ignoring #pragma coffee expression [-Wunknown-pragmas]
           #pragma coffee expression
 
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp: In function ‘void compile_slate(double (*)[21], const double*)’:
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:235:59: error: no matching function for call to ‘subkernel0_cell_to_00_cell_integral_otherwise(Eigen::Matrix<double, 21, 21, 1>&, const double* __restrict__&)’
   subkernel0_cell_to_00_cell_integral_otherwise(T0, coords);
                                                           ^
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:13:21: note: candidate: template<class Derived> void subkernel0_cell_to_00_cell_integral_otherwise(const Eigen::MatrixBase<Derived>&, const double*, const double*)
  static inline void subkernel0_cell_to_00_cell_integral_otherwise (const Eigen::MatrixBase<Derived> &  A_ , const double *restrict coords , const double *restrict cell_sizes )
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:13:21: note:   template argument deduction/substitution failed:
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:235:59: note:   candidate expects 3 arguments, 2 provided
   subkernel0_cell_to_00_cell_integral_otherwise(T0, coords);
                                                           ^
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp: In function ‘PetscErrorCode wrap_compile_slate(int, int, Mat, int32_t*, int32_t*, double*, int32_t*, double*, int32_t*)’:
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:298:62: error: too many arguments to function ‘void compile_slate(double (*)[21], const double*)’
     compile_slate(buffer_arg0_0, buffer_arg1_0, buffer_arg2_0);
                                                              ^
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:229:21: note: declared here
  static inline void compile_slate (double  A2[21][21] , const double *restrict coords )
                     ^~~~~~~~~~~~~
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp: At global scope:
/home/pefarrell/local/firedrake/firedrake-dev-20180914/.cache/pyop2/821d4cb706733265a965d3b8e14473ca_p16841.cpp:257:23: warning: ‘PetscErrorCode apply_extruded_mask(PetscSection, const PetscInt*, int64_t, int, int, int, PetscInt*)’ defined but not used [-Wunused-function]
 static PetscErrorCode apply_extruded_mask(PetscSection section,
                       ^~~~~~~~~~~~~~~~~~~
```

**Comment by thomasgibson** [comment]
I can reproduce the error. A quick glance at the generated TSFC kernel and Slate code seems to indicate that the kernel argument `cell_size` is not being handled. I'll go ahead and fix this.

**Comment by rckirby** [comment]
While I don't doubt there is a slate bug, Argyris, Bell and Morley have no
internal degrees of freedom.  What would be the point of using them with
SLATE?  (One could condense the internal DOF for Hermite and/or implement
higher-order Hermite & Argyris, but I don't think these are in FIAT &
hooked up).

**Comment by wence-** [comment]
In #1309 we use it to compute the cellwise inverse as part of a fortin operator. Rather than using it for static condensation.

**Comment by rckirby** [comment]
Ah, makes sense.

**Comment by thomasgibson** [comment]
Fixed in #1311
